clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -82,6 +95,24 @@
 
     private final Time time;
 
+    private final Map<Integer, Collection<ApiVersionsResponse.ApiVersion>> nodeApiVersions = new HashMap<>();
+
+    private final Queue<String> apiVersionsRequestQueue = new ConcurrentLinkedQueue<>();
```

Comment:
We would need some kind of ds to keep track of api versions request that need to be sent during poll. If we were to use `ClusterConnectionStates` to do this, we will either have to add a state between `CheckingApiVersions` and `Ready` to indicate that api versions check is already in progress for a connection, or we will have to move trigger for sending api versions from poll to handleConnections. I am more in favor of having it the way it is right now, and have the api version requests be sent during polls. However, I do agree that the ds for this can be a set, and I will make that change.